source/renderer/app/components/voting/voting-registration-wizard-steps/VotingRegistrationStepsEnterPinCode.js

```diff
@@ -122,6 +138,37 @@ export default class VotingRegistrationStepsEnterPinCode extends Component<Props
     }
   );
 
+  onChangePinCode = (values: Array<string>, newValue: string, key: number) => {
+    const { form } = this;
+    const pinCodeField = form.$('pinCode');
+    const pinCodeFieldProps = pinCodeField.bind();
+    const { pinValues } = this.state;
+    pinValues[key] = newValue;
+
+    this.setState({
+      pinValues,
+      selectedPinField: 'pinCode',
+    });
+    pinCodeFieldProps.onChange(values);
+  };
+
+  onChangeRepeatPinCode = (
+    values: Array<string>,
+    newValue: string,
+    key: number
+  ) => {
+    const { form } = this;
+    const repeatPinCodeField = form.$('repeatPinCode');
+    const repeatPinCodeFieldProps = repeatPinCodeField.bind();
+    const { repeatPinValues } = this.state;
+    repeatPinValues[key] = newValue;
```

Comment:
The same issue, the state should not be mutated outside the `setState` function

Reply:
Added local variable to avoid this isssue